--- ../src-base/minecraft/net/minecraft/block/BlockRedstoneTorch.java
+++ ../src-work/minecraft/net/minecraft/block/BlockRedstoneTorch.java
@@ -16,7 +16,7 @@
 public class BlockRedstoneTorch extends BlockTorch
 {
     private boolean field_150113_a;
-    private static Map field_150112_b = new HashMap();
+    private static Map field_150112_b = new java.util.WeakHashMap(); // FORGE - fix vanilla MC-101233
     private static final String __OBFID = "CL_00000298";
 
     private boolean func_150111_a(World p_150111_1_, int p_150111_2_, int p_150111_3_, int p_150111_4_, boolean p_150111_5_)
