--- ../src-base/minecraft/net/minecraft/world/storage/SaveHandler.java
+++ ../src-work/minecraft/net/minecraft/world/storage/SaveHandler.java
@@ -5,6 +5,8 @@
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
 import java.io.IOException;

 import cpw.mods.fml.common.FMLCommonHandler;
@@ -81,7 +83,7 @@
        try
        {
            File file1 = new File(this.worldDirectory, "session.lock");
-            DataInputStream datainputstream = new DataInputStream(new FileInputStream(file1));
+            DataInputStream datainputstream = new DataInputStream(new BufferedInputStream(new FileInputStream(file1)));

            try
            {
@@ -118,7 +120,7 @@
        {
            try
            {
-                nbttagcompound = CompressedStreamTools.readCompressed(new FileInputStream(file1));
+                nbttagcompound = CompressedStreamTools.readCompressed(new BufferedInputStream(new FileInputStream(file1)));
                nbttagcompound1 = nbttagcompound.getCompoundTag("Data");
                worldInfo = new WorldInfo(nbttagcompound1);
                FMLCommonHandler.instance().handleWorldDataLoad(this, worldInfo, nbttagcompound);
@@ -141,7 +143,7 @@
        {
            try
            {
-                nbttagcompound = CompressedStreamTools.readCompressed(new FileInputStream(file1));
+                nbttagcompound = CompressedStreamTools.readCompressed(new BufferedInputStream(new FileInputStream(file1)));
                nbttagcompound1 = nbttagcompound.getCompoundTag("Data");
                worldInfo = new WorldInfo(nbttagcompound1);
                FMLCommonHandler.instance().handleWorldDataLoad(this, worldInfo, nbttagcompound);
@@ -256,6 +258,7 @@
             }
 
             file1.renameTo(file2);
+            net.minecraftforge.event.ForgeEventFactory.firePlayerSavingEvent(p_75753_1_, this.field_75771_c, p_75753_1_.func_110124_au().toString());
         }
         catch (Exception exception)
         {
@@ -274,7 +277,7 @@

            if (file1.exists() && file1.isFile())
            {
-                nbttagcompound = CompressedStreamTools.readCompressed(new FileInputStream(file1));
+                nbttagcompound = CompressedStreamTools.readCompressed(new BufferedInputStream(new FileInputStream(file1)));
            }
        }
        catch (Exception exception)
@@ -286,6 +289,7 @@
             p_75752_1_.func_70020_e(nbttagcompound);
         }
 
+        net.minecraftforge.event.ForgeEventFactory.firePlayerLoadingEvent(p_75752_1_, field_75771_c, p_75752_1_.func_110124_au().toString());
         return nbttagcompound;
     }
 
@@ -331,7 +335,7 @@

            if (file1.exists() && file1.isFile())
            {
-                return CompressedStreamTools.readCompressed(new FileInputStream(file1));
+                return CompressedStreamTools.readCompressed(new BufferedInputStream(new FileInputStream(file1)));
            }
        }
        catch (Exception exception)