--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderLightningBolt.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderLightningBolt.java
@@ -24,18 +24,18 @@
         double[] adouble1 = new double[8];
         double d3 = 0.0D;
         double d4 = 0.0D;
-        Random random = new Random(p_76986_1_.field_70264_a);
+        Random random = new org.bogdang.modifications.random.XSTR(p_76986_1_.field_70264_a);
 
         for (int i = 7; i >= 0; --i)
         {
             adouble[i] = d3;
             adouble1[i] = d4;
             d3 += (double)(random.nextInt(11) - 5);
             d4 += (double)(random.nextInt(11) - 5);
         }
 
         for (int k1 = 0; k1 < 4; ++k1)
         {
-            Random random1 = new Random(p_76986_1_.field_70264_a);
+            Random random1 = new org.bogdang.modifications.random.XSTR(p_76986_1_.field_70264_a);
 
             for (int j = 0; j < 3; ++j)
             {