--- ../src-base/minecraft/net/minecraft/world/storage/SaveFormatOld.java
+++ ../src-work/minecraft/net/minecraft/world/storage/SaveFormatOld.java
@@ -4,7 +4,7 @@
import cpw.mods.fml.relauncher.SideOnly;
import java.io.File;
import java.io.FileInputStream;
-import java.io.FileOutputStream;
+import java.io.*;
import java.util.ArrayList;
import java.util.List;
import net.minecraft.client.AnvilConverterException;
@@ -75,7 +75,7 @@
            {
                try
                {
-                    nbttagcompound = CompressedStreamTools.readCompressed(new FileInputStream(file2));
+                    nbttagcompound = CompressedStreamTools.readCompressed(new BufferedInputStream(new FileInputStream(file2)));
                    nbttagcompound1 = nbttagcompound.getCompoundTag("Data");
                    return new WorldInfo(nbttagcompound1);
                }
@@ -91,7 +91,7 @@
            {
                try
                {
-                    nbttagcompound = CompressedStreamTools.readCompressed(new FileInputStream(file2));
+                    nbttagcompound = CompressedStreamTools.readCompressed(new BufferedInputStream(new FileInputStream(file2)));
                    nbttagcompound1 = nbttagcompound.getCompoundTag("Data");
                    return new WorldInfo(nbttagcompound1);
                }
@@ -118,7 +118,7 @@
            {
                try
                {
-                    NBTTagCompound nbttagcompound = CompressedStreamTools.readCompressed(new FileInputStream(file2));
+                    NBTTagCompound nbttagcompound = CompressedStreamTools.readCompressed(new BufferedInputStream(new FileInputStream(file2)));
                    NBTTagCompound nbttagcompound1 = nbttagcompound.getCompoundTag("Data");
                    nbttagcompound1.setString("LevelName", p_75806_2_);
                    CompressedStreamTools.writeCompressed(nbttagcompound, new FileOutputStream(file2));